pkg/sql/schemachanger/scbuild/internal/scbuildstmt/process.go

```diff
@@ -11,31 +11,58 @@
 package scbuildstmt
 
 import (
+	"reflect"
+
+	"github.com/cockroachdb/cockroach/pkg/settings"
 	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scerrors"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 )
 
+var declarativeForSupportedAllOps = settings.RegisterBoolSetting(
+	"sql.schema_changer.declarative_for_all",
+	"when set to true the declarative schema changer is enabled for "+
+		"buth supported/unsupported operations which are full tested.",
+	false,
+)
+
+type supportedStatement struct {
+	fn             interface{}
+	fullySupported bool
+}
+
+// Tracks operations which are fully supported when the declarative schema
+// changer is enabled. Operations marked as non-fully supported can only be
+// with the sql.schema_changer.declarative_for_all toggle enabled.
+var supportStatements = map[reflect.Type]supportedStatement{
+	reflect.TypeOf((*tree.AlterTable)(nil)):   {AlterTable, false},
```

Comment:
This isn't going to work, because we'll want to control this setting at a more granular level than this. ALTER TABLE is just too broad.

I like this idea of a map of `reflect.Type`, though!